packages/cli/src/lib/plugins/clean.ts

```diff
@@ -0,0 +1,506 @@
+ * Determines whether to clean the current project or scan for projects in the directory.
+ * @param options - Clean options that control the cleanup behavior
+ */
+async function cleanCommand(options: CleanOptions) {
+  intro('🧹 RNEF Clean');
+  
+  try {
+    // Try to get the project root (automatically finds the project by traversing up from cwd)
+    const projectRoot = getProjectRoot();
+    // If we get here, we're in an RNEF project
+    await cleanProject(projectRoot, options);
+    outro('Project cleaned successfully!');
+    return;
+  } catch {
+    // Not in an RNEF project, scan for projects in current directory
+    const scanDir = process.cwd();
+    const scanSpinner = spinner();
+    scanSpinner.start('Scanning for RNEF projects...');
+    
+    const projects = findRnefProjects(scanDir);
+    scanSpinner.stop(`Found ${projects.length} RNEF project(s)`);
+    
+    if (projects.length === 0) {
+      outro('No RNEF projects found in the current directory or subdirectories.');
+      return;
+    }
+    
+    if (projects.length === 1) {
+      // Only one project found, ask if user wants to clean it
+      const shouldClean = await promptConfirm({
+        message: `Clean the RNEF project "${projects[0].name}"?`,
+        confirmLabel: 'Yes, clean it',
+        cancelLabel: 'No, cancel',
+      });
+      
+      if (shouldClean) {
+        await cleanProject(projects[0].path, options);
+        outro('Project cleaned successfully!');
+      } else {
+        outro('Cleanup cancelled.');
+      }
+      return;
+    }
+    
+    // Multiple projects found, let user select
+    const choices = projects.map(project => ({
+      value: project.path,
+      label: `${project.name} (${color.dim(project.path)})`,
+    }));
+    
+    const selectedProjects = await promptMultiselect({
+      message: 'Select RNEF projects to clean:',
+      options: choices,
+    });
+    
+    if (selectedProjects.length === 0) {
+      outro('No projects selected for cleanup.');
+      return;
+    }
+    
+    // Clean selected projects
+    for (const projectPath of selectedProjects) {
+      const project = projects.find(p => p.path === projectPath);
+      if (project) {
+        await cleanProject(projectPath, options);
+      }
+    }
+    
+    outro(`Cleaned ${selectedProjects.length} project(s) successfully!`);
+  }
+}
+
+export const cleanPlugin = () => (api: PluginApi): PluginOutput => {
+  api.registerCommand({
+    name: 'clean',
+    description: 'Clean caches and build artifacts for RNEF projects',
+    action: async (options: CleanOptions) => {
+      try {
+        await cleanCommand(options);
+      } catch (error) {
+        if (error instanceof RnefError) {
+          logger.error(error.message);
+          if (error.cause) {
+            logger.debug(`Cause: ${error.cause}`);
+          }
+        } else {
+          logger.error('Unexpected error during cleanup:', error);
+        }
+        process.exit(1);
+      }
```

Comment:
Let's just call the command, when it fails, it should be handled by similar logic in cli.ts file
```suggestion
      await cleanCommand(options);
```